### DISCUSSION TITLE
Mapping over arrays?

### DISCUSSION THREAD

**Discussion by clayrat** [cvc5/cvc5#10797] [Q&A]
Z3 is said to implement its array theory based on the paper ["Generalized, efficient array decision procedures"](http://leodemoura.github.io/files/fmcad09.pdf), which describes an extension of the SMT [ArraysEx theory](https://smt-lib.org/theories-ArraysEx.shtml) to the CAL theory with 2 new operations `K` (constant) and `map`. It seems that CVC5 does support constant arrays, but I couldn't find a way to express the map operation, e.g. [this code from the Z3 documentation](https://microsoft.github.io/z3guide/docs/theories/Arrays/#mapping-functions-on-arrays) throws an `Unknown indexed function 'map'`  error in CVC5:

```
(declare-const a (Array Int Bool))
(declare-const b (Array Int Bool))
(declare-const c (Array Int Bool))
(assert (not (= ((_ map and) a b) ((_ map not) ((_ map or) ((_ map not) b) ((_ map not) a))))))
(check-sat)
```

**Comment by bobismijnnaam** [comment]
I would also be interested in this.
